xbundle/__init__.py

```diff
@@ -1,596 +1,512 @@
-        open(fn, 'w').write(self.pp_xml(xml))
+                # Moved "if" statement after the "open" statement, so we
+                # no longer create zero-byte files here.
+                if fxml.text not in (None, ""):
+                    with open(join(adir, filename), 'w') as output:
+                        output.write(fxml.text)
+            except IOError as err:
+                log.error(
+                    'failed to write about file %s, error %s',
+                    join(adir, filename), err
+                )
+
+    def write_xml_file(self, filename, xml, force_overwrite=False):
+        """
+        Write an XML file to disk.
+        """
+        if (not force_overwrite) and (
+                xml.tag in self.no_overwrite) and exists(filename):
+            log.debug("Not overwriting %s for %s", filename, xml)
+            filename = filename + '.new'
+        with open(filename, 'w') as output:
+            output.write(pp_xml(xml))
 
     def export_xml_to_directory(self, elem, dowrite=False):
-        '''
-        Do this recursively.  If an element is a descriptor, then put that in its own
-        subdirectory.
-        '''
-        def write_xml(x):
-            un = x.get('url_name')
-            if un is None:
-                self.errlog("Oops!  error in export_xml_to_directory, missing url_name:")
-                self.errlog(x)
+        """
+        Do this recursively.  If an element is a descriptor,
+        then put that in its own subdirectory.
+        """
+        def write_xml(element):
+            """
+            Write XML file from an XML element.
+            """
+            url_name = element.get('url_name')
+            if url_name is None:
+                log.error("missing url_name: %s", element)
             elem.attrib.pop('url_name')
             if 'url_name_orig' in elem.attrib and self.keep_urls:
                 elem.attrib.pop('url_name_orig')
-            edir = self.mkdir(join(self.dir, x.tag))
-            self.write_xml_file(join(edir, un + '.xml'), x)
-            return un
+            edir = mkdir(join(self.path, element.tag))
+            self.write_xml_file(join(edir, url_name + '.xml'), element)
+            return url_name
 
-        # print elem
         if elem.tag == 'descriptor':
-            # print "--> %s" % list(elem)
-            self.export_xml_to_directory(elem[0], dowrite=True)  # recurse on children, depth first
-            elem.tag = elem.get('tag')			# change descriptor to point to new elem
+            # Recurse on children, depth first.
+            self.export_xml_to_directory(elem[0], dowrite=True)
+            # Change descriptor to point to new elem.
+            elem.tag = elem.get('tag')
             elem.set('url_name', elem.get('url_name'))
             elem.attrib.pop('tag')
-            # self.export_xml_to_directory(elem)	# recurse on this tag
-
-        elif elem.tag == etree.Comment:			# comment <!-- foo -->
-            pass
```

Comment:
I don't understand the "keeping comments" question. The code removed here was either commented out, or contained nothing but `pass`.